### PR TITLE
Add Jlama documentation

### PR DESCRIPTION
**Pull request by mariofusco**
@geoand please review.

### CODE REVIEW

**Comment by geoand** [comment]
Thanks a lot!

I'll check tomorrow

**Comment by geoand** [comment]
I merged #1017 which now means that this needs a rebase (and potentially some changed in the content).

**Comment by geoand** [comment]
@gsmet can you please help out with @mariofusco's question?

**Comment by mariofusco** [comment]
@gsmet @geoand I've rebase and updated this pull request according with the latest changes, but my doubts regarding the current state remain. Please review and let me know if there is something missing.

File: docs/modules/ROOT/pages/jlama.adoc [hunk 1]
```diff
@@ -0,0 +1,73 @@
+= Jlama
+
+include::./includes/attributes.adoc[]
+
+https://github.com/tjake/Jlama[Jlama] provides a way to run large language models (LLMs) locally and in pure Java, eventually also embedded in your Quarkus application.
```
Comment (geoand) [inline docs/modules/ROOT/pages/jlama.adoc:5]: What "eventually"?

File: docs/modules/ROOT/pages/jlama.adoc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+You can run many https://huggingface.co/tjake[models] such as LLama3, Mistral, Granite and many others on your machine.
+
+[#_prerequisites]
+== Prerequisites
+
+To use Jlama it is necessary to run on Java 20 or later. This is because it utilizes the new https://openjdk.org/jeps/448[Vector API] for faster inference. Note that the Vector API is still a Java preview features, so it is required to explicitly enable it.
```
Comment (geoand) [inline docs/modules/ROOT/pages/jlama.adoc:11]: Our extension actually requires 21

**Review by geoand — COMMENTED** [review]
Thanks! I added some small questions

File: docs/modules/ROOT/pages/jlama.adoc (hunk 1 above)
Comment (mariofusco) [reply docs/modules/ROOT/pages/jlama.adoc]: You're right at the moment is the only why to make it work, I rephrased accordingly. I wonder if we could also talk via REST to an external server running Jlama and if this use case would actually make sense.

Comment (geoand) [reply docs/modules/ROOT/pages/jlama.adoc]: We haven't tried it, but assuming that it implements the OpenAI protocol correctly, it should just work

**Review by geoand — APPROVED** [review]
Thanks!

**Comment by geoand** [comment]
We might also want to mention that limitations of dev-mode and what flags are needed

**Comment by mariofusco** [comment]
> We might also want to mention that limitations of dev-mode and what flags are needed

Ah right, I forgot to mention the necessary flags to enable the Vector API. Give me one minute to add them before merging.

**Comment by geoand** [comment]
Thanks!

Also that in dev-mode C2 is not enabled, but that will be fixed with Quarkus 3.17 when `<forceC2>true</forceC2>` is set

**Comment by mariofusco** [comment]
@geoand Done.

File: docs/modules/ROOT/pages/jlama.adoc [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+----
+<configuration>
+    <jvmArgs>--enable-preview --enable-native-access=ALL-UNNAMED</jvmArgs>
+    <modules>
+        <module>jdk.incubator.vector</module>
+    </modules>
+</configuration>
+----
+
+=== Dev Service
```
Comment (geoand) [inline docs/modules/ROOT/pages/jlama.adoc:32]:
```suggestion
=== Dev Mode
```

**Comment by geoand** [comment]
One final squash and we're good to go

**Comment by mariofusco** [comment]
> One final squash and we're good to go

Done.

**Review by geoand — APPROVED** [review]
Thanks!

**Comment by geoand** [comment]
cc @lordofthejars in case you want to add the relevant Llama3.java docs

**Comment by lordofthejars** [comment]
Oh yes I will work next week when CR2 is released.

**Comment by geoand** [comment]
Release is on the way

**Comment by geoand** [comment]
https://github.com/quarkiverse/quarkus-langchain4j/releases/tag/0.21.0.CR3 was just released

**Comment by geoand** [comment]
@lordofthejars do you still plan to add docs for Llama3?

**Pull request merged**